test/metabase/api/collection_test.clj

```diff
@@ -1494,6 +1494,250 @@
                           remove-non-personal-collections
                           mt/boolean-ids-and-timestamps)))))))
 
+(deftest ^:parallel dashboard-question-candidates-test
+  (testing "GET /api/collection/:id/dashboard-question-candidates"
+    (testing "Card is in single dashboard"
+      (mt/with-temp [:model/Collection {coll-id :id} {}
+                     :model/Dashboard {dash-id :id} {:collection_id coll-id}
+                     :model/Card {card-id :id} {:collection_id coll-id}
+                     :model/DashboardCard _ {:dashboard_id dash-id :card_id card-id}]
+        (is (= #{card-id}
+               (->> (mt/user-http-request :crowberto :get 200 (str "collection/" coll-id "/dashboard-question-candidates"))
+                    :data
+                    (map :id)
+                    (into #{}))))
+        (is (= #{:id :name :description :sole_dashboard_info}
+               (->> (mt/user-http-request :crowberto :get 200 (str "collection/" coll-id "/dashboard-question-candidates"))
+                    :data
+                    first
+                    keys
+                    (into #{}))))
+        (is (= #{:id :name :description}
+               (->> (mt/user-http-request :crowberto :get 200 (str "collection/" coll-id "/dashboard-question-candidates"))
+                    :data
+                    first
+                    :sole_dashboard_info
+                    keys
+                    (into #{}))))))
+
+    (testing "Card is in two dashboards"
```

Comment:
These could easily be several separate tests. If they're not sharing state, it's better to break out logically separate test cases into separate `deftest`s, that way if they fail they're less of a headache to debug and so we can parallelize them further

Reply:
Good call - thanks!